test/merge-patient-lab-iips-mocha.js

```diff
@@ -5,151 +5,260 @@ var chaiAsPromised = require('chai-as-promised');
+                .then(commonTest.createComponent).then(function(component){
+
+                    stubState( component );  // REMOVE WHEN STATES DONE
+                    commonTest.sendData( component, 'patient',
+                                         {id: '001',  name: 'Alice', dob: '1979-01-23' });
                     return component;
-                }).then(promiseLater).then(function(component){
-                    return component.rpf.createVni('').getOutputState();
-                }).then(_.property('data')).should.become({id: '001',  name: 'Alice', dob: '1979-01-23', glucose: '75',  date: '2012-02-01'});
-            });
+
+                }).then(commonStubs.promiseLater).then(function(component){
+                    return component.rpf.vni('').inputState('patient');
+
+                }).then(_.property('data')).should.become({id: '001',  name: 'Alice', dob: '1979-01-23' });
+        });
+
+        it("should have labwork input state after input", function() {
+            return Promise.resolve(componentFactory.getComponent )
+                .then(commonTest.createComponent).then(function(component){
+
+                    stubState( component );  // REMOVE WHEN STATES DONE
+                    commonTest.sendData( component, 'labwork',
+                                         {id: '001',  glucose: '75',  date: '2012-02-01'});
+                    return component;
+
+                }).then(commonStubs.promiseLater).then(function(component){
+
+                    return component.rpf.vni('').inputState('labwork');
+
+                }).then(_.keys).then(_.sortBy).should.become(_.sortBy(['data','lm', 'previousLms', 'port']));
+        });
+
+        it("should have patient input state data after input", function() {
+            return Promise.resolve(componentFactory.getComponent )
+                .then(commonTest.createComponent).then(function(component){
+
+                    stubState( component );  // REMOVE WHEN STATES DONE
+                    commonTest.sendData( component, 'labwork',
+                                         {id: '001',  glucose: '75',  date: '2012-02-01'});
+                    return component;
+
+                }).then(commonStubs.promiseLater).then(function(component){
+                    return component.rpf.vni('').inputState('labwork');
+
+                }).then(_.property('data')).should.become({id: '001',  glucose: '75',  date: '2012-02-01'});
+        });
+
+        it.skip( "should have patient and labwork output state after input ports processing", function(done) {
+            return Promise.resolve(componentFactory.getComponent )
+                .then(commonTest.createComponent).then(function(component){
+
+                    sinon.stub(component.outPorts.output, 'send', function( data ) {
+                        data.should.exist;
+                        data.should.not.be.empty;
+                        data.should.have.ownProperty('output');
```

Comment:
This test looks outdated.  It  looks partially like it is trying to test the return value of an updater function and partially like it is trying to test the value of the resulting output State object.